src/shared/test/integration/rpc-bridge-test.js

```diff
@@ -0,0 +1,206 @@
+import Bridge from '../../bridge';
+
+describe('RPC-Bridge integration', () => {
+  const sandbox = sinon.createSandbox();
+  let clock;
+  let port1;
+  let port2;
+  let bridges = [];
+
+  function createBridge() {
+    const bridge = new Bridge();
+    bridges.push(bridge);
+    return bridge;
+  }
+
+  function waitForMessageDelivery() {
+    return new Promise(resolve => setTimeout(resolve, 0));
+  }
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+    const channel = new MessageChannel();
+    port1 = channel.port1;
+    port2 = channel.port2;
+  });
+
+  afterEach(() => {
+    bridges.forEach(bridge => bridge.destroy());
+    sandbox.restore();
+    clock.restore();
+  });
+
+  context('establishing a connection', () => {
+    it('should invoke Bridge `onConnect` callbacks after connecting', async () => {
+      clock.restore();
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+
+      let callbackCount = 0;
+      const callback = () => {
+        ++callbackCount;
+      };
+
+      bridge.onConnect(callback);
+      bridge.onConnect(callback); // allows multiple callbacks to be registered
+      reciprocalBridge.onConnect(callback);
+
+      const channel = bridge.createChannel(port1);
+      reciprocalBridge.createChannel(port2);
+
+      await waitForMessageDelivery();
+      assert.equal(callbackCount, 3);
+
+      // Additional calls to the RPC `connect` method are ignored
+      channel.call('connect');
+      reciprocalBridge.call('connect');
+
+      await waitForMessageDelivery();
+      assert.equal(callbackCount, 3);
+    });
+  });
+
+  context('sending and receiving RPC messages', () => {
+    let bridge;
+
+    beforeEach(() => {
+      bridge = createBridge();
+      const otherChannel = new MessageChannel();
+      bridge.createChannel(port1);
+      bridge.createChannel(otherChannel.port1);
+      const reciprocalBridge1 = createBridge();
+      const reciprocalBridge2 = createBridge();
+      reciprocalBridge1.on('method1', (arg, cb) => cb(null, `${arg}foo`));
+      reciprocalBridge2.on('method1', (arg, cb) => cb(null, `${arg}bar`));
+      reciprocalBridge1.createChannel(port2);
+      reciprocalBridge2.createChannel(otherChannel.port2);
+    });
+
+    it('should invoke Bridge method handler on every channel when calling a RPC method (with Promise)', async () => {
+      const results = await bridge.call('method1', 'params1');
+      assert.deepEqual(results.sort(), ['params1foo', 'params1bar'].sort());
+    });
+
+    it('should invoke Bridge method handler on every channel when calling a RPC method (with callback)', done => {
```

Comment:
This is out of scope for this PR - but could we get rid of support for result callbacks in the `Bridge`? It looks to me like it duplicates the Promise return functionality and the Promise-based API is more convenient for consumers and easier to specify the type of.

Reply:
I am fine removing this test. It was just to demonstrated that the `error` is `null`, which is only possible to be tested in callback-based API (it is functionally the same as no error raised in the Promise-based).